lib/sisimai/bite/email/activehunter.rb

```diff
@@ -0,0 +1,124 @@
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'TransWARE Active!hunter'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-AHMAILID']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from TransWARE Active!hunter
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['x-ahmailid']
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            #  ----- The following addresses had permanent fatal errors -----
+            #
+            # >>> kijitora@example.org <kijitora@example.org>
+            #
+            #  ----- Transcript of session follows -----
+            # 550 sorry, no mailbox here by that name (#5.1.1 - chkusr)
+            v = dscontents[-1]
+
+            if cv = e.match(/\A[>]{3}[ \t]+.+[<]([^ ]+?[@][^ ]+?)[>]\z/)
```

Comment:
__[rubocop]__
* Style/NumericPredicate: Use `(readcursor & Indicators[:deliverystatus]).zero?` instead of `readcursor & Indicators[:deliverystatus] == 0`. [:ref](https://github.com/bbatsov/ruby-style-guide#predicate-methods)

<!-- Comment by [SideCI](https://sideci.com) -->